src/ansys/fluent/core/launcher/launcher_utils.py

```diff
@@ -153,12 +153,27 @@ def get_mode(mode: str) -> "FluentMode":
             If an unknown mode is passed.
         """
         for m in FluentMode:
-            if mode == m.value[0]:
+            if mode == m.value._internal_name:
                 return m
-        else:
-            raise DisallowedValuesError(
-                "mode", mode, ["meshing", "pure-meshing", "solver", "solver-icing"]
-            )
+        raise DisallowedValuesError(
+            "mode", mode, ["meshing", "pure-meshing", "solver", "solver-icing"]
```

Comment:
I didn't notice this list previously. These items are already known, and this will break in maintenance.

Reply:
@seanpearsonuk, shall we not pass the allowed values then?

Reply:
All I mean here is that this is a list that is effectively already known: it can be evaluated based on the enum, correct? If we hard-code it then it can get out of step with the correct value.

Reply:
What about somethig like:
allowed_modes = []
        for m in FluentMode:
            allowed_modes.append(m.value._internal_name)
            if mode == m.value._internal_name:
                return m
        raise DisallowedValuesError(
            "mode", mode, allowed_modes
        )

Reply:
Anything like that as long as it works.

Reply:
@hpohekar, actually we are already looping over these once. So, I didn't create another loop.